pkg/nsx/services/firewall.go

```diff
@@ -1033,13 +1034,31 @@ func (service *SecurityPolicyService) updatePeerExpressions(obj *v1alpha1.Securi
 	}
 	if peer.NamespaceSelector != nil {
 		if !mixedNsSelector {
+			if peer.NamespaceSelector.Size() == 0 {
+				// Since expressions list in model.Group must follow criteria from NSXT:
+				// 1. A non-empty expression list, must be of odd size
+				// 2. An expression list size is equal to or greater than 3
+				// 3. In a list, with indices starting from 0, all non-conjunction expressions must be at even indices
+				// Hence, add one more SegmentPort member condition to meet the criteria aforementioned
+				service.addOperatorIfNeeded(expressions, "AND")
+				clusterSegPortExpression := service.buildExpression(
+					"Condition", "SegmentPort",
```

Comment:
I've tested if putting a duplicated Segment Cluster tag condition here, NSX will  report a "Duplicate expressions specified" error. So, SegmentPort member type is right way.

Also, I've tried to remove NestedExpression from  expressions of group, but it has to be an expression list to meet ceratain criteria, so this workaround fix is only way if we continue to user this group structure.